### PR TITLE
Pin damacus/github-repository module to version 0.0.3

### PR DESCRIPTION
**Pull request by damacus**
- Use module from damacus/github-repository
- Rename our workflow to match what's defined in the module

Signed-off-by: Dan Webb <dan.webb@damacus.io>

# Description

Describe what this change achieves

## Issues Resolved

This is the first step in upgrading to the new module and Terraform version

## Check List

- [ ] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by damacus** [comment]
Things to note,

Terraform repositories have changed like this

```shell
              - "ci / json",
              - "ci / mdl",
              - "ci / yamllint",
              + "lint-unit / mdl",
              + "lint-unit / yamllint",
```

Most repositories now all auto merging

```shell
allow_auto_merge            = false -> true
```

File: main.tf [hunk 1]
```diff
@@ -1,6 +1,9 @@
 module "repository" {
-  for_each                  = { for repo in var.repository : repo.name => repo }
-  source                    = "./modules/repository"
+  for_each = { for repo in var.repository : repo.name => repo }
+
+  source  = "damacus/github-repository/module"
+  version = "0.0.3"
```
Comment (ramereth) [inline main.tf:5]: Can you please elaborate on why this change is needed?

Comment (damacus) [reply main.tf:5]: Having it as an external module allows us to version pin, I'll fix the module separately. 

I'm also going to do the same thing over at test-kitchen some time soon

Comment (ramereth) [reply main.tf:5]: But where does this source actually exist? I'm still a little confused
